### PR TITLE
[risk=low][RW-13484] Add initialCreditsExpirationEpochMillis to Workspace endpoints

### PR DESCRIPTION
**Pull request by jmthibault79**
Return this value (when applicable) in the user Profile object and in the Workspace object for at least these API calls
* ~Get Profile~ moved to #8743
* Get Workspace
* List Workspaces
* List Recent Workspaces
* List Featured Workspaces (new style)

Tested locally

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [x] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

**Review by yonghaoy — APPROVED** [review]

**Review by yonghaoy — CHANGES_REQUESTED** [review]
We also need bypassed field from DB table. 
I believe UI would also need that field to determine expired or not.

**Comment by jmthibault79** [comment]
re: bypassed

Great catch, thank you.  Since we will need to check user-bypassed as well as institution-bypassed to decide if there is a valid expiration time, then I don't think we can use the DB field as a sole source of truth for this.  IMO it may be time to make a small Service here.

**Comment by evrii** [comment]
> re: bypassed
> 
> Great catch, thank you. Since we will need to check user-bypassed as well as institution-bypassed to decide if there is a valid expiration time, then I don't think we can use the DB field as a sole source of truth for this. IMO it may be time to make a small Service here.

I've started a service in my cron job work. I can break that out into a small PR, so we are working from a similar base.

File: api/src/test/java/org/pmiops/workbench/featuredworkspace/FeaturedWorkspaceTest.java [hunk 1]
```diff
@@ -44,11 +43,9 @@ public class FeaturedWorkspaceTest {
   @MockBean private FeaturedWorkspaceDao mockFeaturedWorkspaceDao;
   @MockBean private FeaturedWorkspaceMapper mockFeaturedWorkspaceMapper;
   @MockBean private FireCloudService mockFireCloudService;
+  @MockBean private InitialCreditsExpirationService mockInitialCreditsExpirationService;
   @MockBean private WorkspaceMapper mockWorkspaceMapper;
 
-  // Delete this once published flag is on
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/featuredworkspace/FeaturedWorkspaceTest.java:49]: unrelated - but we can do this now

File: api/src/test/java/org/pmiops/workbench/featuredworkspace/FeaturedWorkspaceTest.java [hunk 2]
```diff
@@ -99,17 +94,17 @@ public void testFeaturedCategory_getFeaturedCategoryForFeaturedWorkspace() {
   }
 
   void mockFeaturedWorkspaces(String namespace, DbFeaturedCategory dbFeaturedCategory) {
-    DbWorkspace mockdbWorkspace =
+    DbWorkspace mockDbWorkspace =
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/featuredworkspace/FeaturedWorkspaceTest.java:97]: just capitalization

File: api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java [hunk 3]
```diff
@@ -540,11 +550,12 @@ public ResponseEntity<Workspace> updateWorkspace(
...
     }
 
-    final Workspace editedWorkspace = workspaceMapper.toApiWorkspace(dbWorkspace, fcWorkspace);
+    final Workspace editedWorkspace =
+        workspaceMapper.toApiWorkspace(dbWorkspace, fcWorkspace, initialCreditsExpirationService);
 
     workspaceAuditor.fireEditAction(
         originalWorkspace, editedWorkspace, dbWorkspace.getWorkspaceId());
-    return ResponseEntity.ok(workspaceMapper.toApiWorkspace(dbWorkspace, fcWorkspace));
+    return ResponseEntity.ok(editedWorkspace);
```
Comment (evrii) [inline api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java:558]: This seems like a good idea. I am curious though, do you know why this was being mapped twice?

**Review by evrii — APPROVED** [review]
Tested locally and everything worked as expected. The code looks good.

File: api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java (hunk 3 above)
Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java:558]: I think that was just an oversight

**Review by yonghaoy — APPROVED** [review]

**Pull request merged**